### ISSUE TITLE
createLazyFileRoute function doesn't have validateSearch property

### ISSUE THREAD

**Issue by skaionedev** [TanStack/router#1209]
### Describe the bug

I can't use validateSearch property for lazy file routes as it's for regular file routes
![Screenshot 2024-02-22 at 16 39 03](https://github.com/TanStack/router/assets/75416964/453c27b7-4eb7-4692-ab1f-dd5b3ca75c41)

### Steps to Reproduce the Bug or Issue

1. create a fresh vite + react app with router
2. create lazy file route (e.g. about.lazy.tsx)

### Expected behavior

I expect lazy file routes have same properties as regular file routes

### Platform

- OS: macOS
- Browser: Brave
- Version:  1.62.165

**Comment by SeanCassiere** [comment]
The `createLazyFileRoute` helper is meant to be used for routes that are being code-split. For example `src/routes/about.lazy.tsx`. This configuration only lets you define the components for a specific Route that are then imported at runtime using code-splitting.

When you need to define the Route configuration (such as configuring the loader, beforeLoad, validateSearch, etc.), it is done in a separate file which would be named as `src/routes/about.tsx`. In here, you would use the `createFileRoute` helper to define the Route configuration itself.

```tsx
import { createFileRoute } from '@tanstack/react-router'
import { z } from 'zod'
export const Route = createFileRoute("/about")({
  validateSearch: (search) => z.object({ foo: z.string().catch("bar") }).parse(search)
});
```

**TLDR;** You'll need both `about.lazy.tsx` AND `about.tsx` if you want the *about* route to be code-split whilst utilizing the extensible Route configuration.

**Comment by ShellWen** [comment]
It should be pinned, lots of developers are very confused about this feature.

**Comment by SeanCassiere** [comment]
> It should be pinned, lots of developers are very confused about this feature.

This has been stated in docs [over here](https://tanstack.com/router/latest/docs/framework/react/guide/code-splitting#how-does-tanstack-router-split-code), but it's easy to understand why someone could miss it. Tanstack Router covers a lot, and the fact that virtual routes *"just work"* means people don't see fully read the page on code-splitting and how it works.

IMO: Tanstack Router is definitely not the most beginner friendly solution at the moment, and honestly I don't think it ever will be given the kinds of complex stuff it aims to do.

But we should see alot of the documentation pages getting a cleanup when we begin the push for v2.

**Comment by skaionedev** [comment]
@SeanCassiere  Thank you 🙏
